core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -1274,20 +1275,20 @@ public void testSetStatistics() {
     Assert.assertEquals("Statistics file snapshot", 43L, statisticsFile.snapshotId());
     Assert.assertEquals("Statistics file path", "/some/path/to/stats/file", statisticsFile.path());
 
-    TableMetadata withStatisticsReplaced =
+    TableMetadata withStatisticsAppended =
         TableMetadata.buildFrom(withStatistics)
             .setStatistics(
                 43,
                 new GenericStatisticsFile(
                     43, "/some/path/to/stats/file2", 128, 27, ImmutableList.of()))
             .build();
 
-    Assertions.assertThat(withStatisticsReplaced.statisticsFiles())
-        .as("There should be one statistics file registered")
-        .hasSize(1);
-    statisticsFile = Iterables.getOnlyElement(withStatisticsReplaced.statisticsFiles());
-    Assert.assertEquals("Statistics file snapshot", 43L, statisticsFile.snapshotId());
-    Assert.assertEquals("Statistics file path", "/some/path/to/stats/file2", statisticsFile.path());
+    Assertions.assertThat(withStatisticsAppended.statisticsFiles())
+        .as("There should be two statistics files registered")
+        .hasSize(2)
```

Comment:
As I fixed the overwrite to append. There will be two stats file for this snapshot id. Hence, updated the testcases.

Reply:
I think this is incorrect.